### PR TITLE
GraphMakie update 

### PR DESCRIPTION
**Pull request by vyudu**
Refactoring the Graphviz code into GraphMakie. Some plots generated: 

Brusselator species-reaction graph
fig1
Brusselator complex graph
fig2
Repressilator species-reaction graph
fig3

Might be worth playing around with the layouts a bit more to see if we can make the edge-lengths for these graphs more consistent and have the layouts be more horizontal. Will work on updating the documentation in a follow-up.

### CODE REVIEW

**Comment by isaacsas** [comment]
The first and third look great! The second is a bit strange. Do you want to tweak the layouts more in the PR, or get it merged as is and update later? (I ask as I'll review it sonner in the latter case or wait till you say you are done in the former case).

**Comment by isaacsas** [comment]
Also, is this based on the current master? I thought I had removed BifurcationKit there so the docs build but this seems to be crashing on it.

**Comment by vyudu** [comment]
Probably just merge this one as is and I'll look more at it in a follow-up. Also just rebased it, it seems like the remote branch was based on an earlier master commit

**Comment by isaacsas** [comment]
It may still fail -- I have another PR that is completely removing Bifkit, so it may need one more rebasing once that is merged (but if tests pass and docs build as is I'll merge after reviewing).

**Review by isaacsas — CHANGES_REQUESTED** [review]
Is there any way to add some tests for this?

File: ext/CatalystGraphMakieExtension.jl
Comment (vyudu) [reply ext/CatalystGraphMakieExtension.jl:7]: Some of these imports were there before I believe. Though I thought it'd make it easier to not have to write out Graphs.SimpleEdge, etc. too much in the code since some of these names aren't exported by default, but I can just have them be qualified in the code, no issue there

Comment (isaacsas) [reply ext/CatalystGraphMakieExtension.jl:7]: Just use `using` then. (We should eventually get rid of import everywhere anyways I think, but that is not high priority).

**Comment by isaacsas** [comment]
@vyudu probably worth updating this to master now that the BifurcationKit PR is merged. That should hopefully have fixed some issues in tests and doc builds.

File: ext/CatalystGraphMakieExtension.jl
Comment (isaacsas) [inline ext/CatalystGraphMakieExtension.jl:12]: In Catalyst.jl have

```julia
function plot_speciesreaction_graph end
function plot_complex_graph end
export plot_speciesreaction_graph, plot_complex_graph
```
Note you also need to update the doc strings below for the new names.

Comment (isaacsas) [reply ext/CatalystGraphMakieExtension.jl:12]: Actually, for consistency it should probably be
```julia
plot_species_reaction_graph
```

Comment (isaacsas) [reply ext/CatalystGraphMakieExtension.jl:12]: Which admittedly is very long. Can we think of a shorter descriptive name?

Comment (vyudu) [reply ext/CatalystGraphMakieExtension.jl:12]: plot_sr_graph?

Comment (isaacsas) [reply ext/CatalystGraphMakieExtension.jl:12]: What about just `plot_network(...)` and `plot_complexes(...)`?

File: ext/CatalystGraphMakieExtension/rn_graph_plot.jl
Comment (vyudu) [reply ext/CatalystGraphMakieExtension/rn_graph_plot.jl:69]: Just thought some people might know them better as Petri nets, but yeah probably better to mention this in the documentation than have another name.

**Review by isaacsas — APPROVED** [review]

**Comment by isaacsas** [comment]
@vyudu feel free to merge when done now.

**Comment by isaacsas** [comment]
@vyudu ping -- are you done with this? Can we merge?

**Comment by vyudu** [comment]
I am yeah, we can merge

**Pull request merged**
